AwsCryptographyPrimitives/runtimes/python/src/aws_cryptography_primitives/internaldafny/extern/Signature.py

```diff
@@ -0,0 +1,342 @@
+from cryptography.exceptions import (
+  InvalidSignature,
+  UnsupportedAlgorithm
+)
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.serialization import (
+  NoEncryption,
+  Encoding,
+  PrivateFormat,
+  load_pem_private_key,
+)
+import cryptography.hazmat.primitives.asymmetric.ec as ec
+from cryptography.hazmat.primitives.asymmetric.ec import (
+  SECP256R1,
+  SECP384R1,
+)
+from cryptography.hazmat.primitives.asymmetric.utils import (
+  decode_dss_signature,
+  encode_dss_signature
+)
+
+from collections import namedtuple
+import _dafny
+from enum import Enum
+
+import standard_library.internaldafny.generated.Wrappers as Wrappers
+import aws_cryptography_primitives.internaldafny.generated.ExternDigest as ExternDigest
+from aws_cryptography_primitives.internaldafny.generated.Signature import *
+import aws_cryptography_primitives.internaldafny.generated.Signature
+from aws_cryptography_primitives.internaldafny.generated.AwsCryptographyPrimitivesTypes import (
+  DigestAlgorithm_SHA__256,
+  DigestAlgorithm_SHA__384,
+  Error_AwsCryptographicPrimitivesError
+)
+
+
+# Extend generated class
+class default__(aws_cryptography_primitives.internaldafny.generated.Signature.default__):
+  class ECDSA:
+
+    @staticmethod
+    def ExternKeyGen(signature_algorithm):
+      maybe_signature_algorithm = SignatureAlgorithms.signatureAlgorithm(signature_algorithm)
+      if maybe_signature_algorithm.is_Failure:
+        return Wrappers.Result_Failure(maybe_signature_algorithm.error)
+      
+      private_key = ec.generate_private_key(
+        maybe_signature_algorithm.value.value.curve
+      )
+
+      private_key_pem_bytes = private_key.private_bytes(Encoding.PEM, PrivateFormat.PKCS8, NoEncryption())
+      public_key_pem_bytes = _ecc_encode_compressed_point(private_key)
+
+      return Wrappers.Result_Success(SignatureKeyPair_SignatureKeyPair(
+          verificationKey=_dafny.Seq(public_key_pem_bytes),
+          signingKey=_dafny.Seq(private_key_pem_bytes)
+      ))
+
+    @staticmethod
+    def Sign(signature_algorithm, signing_key, message):
+      maybe_signature_algorithm = SignatureAlgorithms.signatureAlgorithm(signature_algorithm)
+      if maybe_signature_algorithm.is_Failure:
+        return Wrappers.Result_Failure(maybe_signature_algorithm.error)
+      
+      try:
+        private_key = load_pem_private_key(bytes(signing_key), None)
+      except ValueError:
+        return Wrappers.Result_Failure(Error_AwsCryptographicPrimitivesError(
+          message="Could not decode signing_key"
+        ))
+      except TypeError:
+        return Wrappers.Result_Failure(Error_AwsCryptographicPrimitivesError(
+          message="Provided signing_key is encrypted, but encrypted signing_keys are not supported"
```

Comment:
Nit: I think pyca supports BoringSSL and one day may even support AWS-LC. 

So I might not say OpenSSL, as we not certain what the lib-crypto under the hood is.